kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/parquet/ParquetRecordWriterProvider.java

```diff
@@ -57,19 +68,33 @@ public String getExtension() {
   @Override
   public RecordWriter getRecordWriter(final S3SinkConnectorConfig conf, final String filename) {
     return new RecordWriter() {
+      final String adjustedFilename = getAdjustedFilename(filename, getExtension());
       Schema schema = null;
       ParquetWriter<GenericRecord> writer;
       S3ParquetOutputFile s3ParquetOutputFile;
 
       @Override
       public void write(SinkRecord record) {
+        boolean schemaPadded = false;
         if (schema == null) {
-          schema = record.valueSchema();
+          schema = recordView.getViewSchema(record);
+          // pad the schema for AvroParquetWriter that only accepts record types
```

Comment:
This special handling here is also awkward. Is it possible to do these in the corresponding `RecordView` implementations instead?

Reply:
Do you mean the padding logic?

Reply:
Yeah. Doing this one-off handling in parquet record writer is awkward since it has nothing to do with writing parquet records per-say. Instead, it is related to what "view" of the record we want the parquet record writer to see and write. I think this could be done in the `RecordView` by modifying the interface to allow for specifying if the view should be padded or not. i.e.
```
public interface RecordView {
  Schema getViewSchema(SinkRecord record, boolean enveloped);
  Object getView(SinkRecord record, boolean enveloped);
}
```
Also, just a nit: I think of this sort of wrapper object as an envelope. Padding I think of extra empty bytes in a byte[].

Reply:
Yes that can work, for some we may not need to use the `boolean` at all because we don't want to envelop, but that's not an issue. I'll update the PR.